packages/theme-check-common/src/checks/liquid-html-syntax-error/checks/InvalidConditionalNode.ts

```
@@ -0,0 +1,144 @@
+  if (!issue) return;
+
+  const openingTagRange = node.blockStartPosition || node.position;
+  const openingTag = node.source.slice(openingTagRange.start, openingTagRange.end);
+  const markupOffsetInOpening = openingTag.indexOf(markup);
+  if (markupOffsetInOpening < 0) return;
+
+  const startIndex = openingTagRange.start + markupOffsetInOpening;
+  const endIndex = startIndex + markup.length;
+
+  return {
+    message: `Liquid lax parsing issue: ${issue.message}`,
+    startIndex,
+    endIndex,
+    fix: (corrector) => {
+      corrector.replace(startIndex, endIndex, issue.fix);
+    },
+  };
+}
+
+function isValueToken(token: Token): boolean {
+  return token.type === 'literal' || token.type === 'variable';
+}
+
+function isOperatorToken(token: Token): boolean {
+  return token.type === 'logical' || token.type === 'comparison';
+}
+
+function checkInvalidStartingToken(tokens: Token[]): ExpressionIssue | null {
+  const firstToken = tokens[0];
+  if (firstToken.type === 'invalid' || firstToken.type === 'comparison') {
+    return {
+      message: `Malformed expression starting with invalid token '${firstToken.value}'`,
+      fix: 'false',
+    };
+  }
+  return null;
+}
+
+function checkTrailingTokensAfterComparison(tokens: Token[]): ExpressionIssue | null {
+  const COMPARISON_LENGTH = 3;
+  const minTokensForTrailing = COMPARISON_LENGTH + 1;
+
+  for (let i = 0; i <= tokens.length - minTokensForTrailing; i++) {
+    const [v1, op, v2] = tokens.slice(i, i + 3);
+    const remaining = tokens.slice(i + 3);
+
+    if (isValueToken(v1) && op.type === 'comparison' && isValueToken(v2)) {
+      if (remaining.length > 0) {
+        if (remaining[0].type !== 'logical') {
+          const validExpr = tokens.slice(0, i + 3).map(t => t.value).join(' ');
+          const junk = remaining.map(t => t.value).join(' ');
+          return {
+            message: `Trailing tokens ignored after comparison: '${junk.trim()}'`,
```

@karreiro we should all should unify how we want to report these error messages. For boolean expressions, we have a generic messaged like "Syntax is unsupported". Do we want that for ALL liquid syntax errors?